lib/spack/spack/installer.py

```diff
@@ -1878,61 +1879,80 @@ def _real_install(self):
             # everything to log_path, and provide the phase for logging
-            for i, (phase_name, phase_attr) in enumerate(zip(
-                    pkg.phases, pkg._InstallPhase_phases)):
+            if sys.platform != 'win32':
+                for i, (phase_name, phase_attr) in enumerate(zip(
+                        pkg.phases, pkg._InstallPhase_phases)):
+
+                    # Keep a log file for each phase
+                    log_dir = os.path.dirname(pkg.log_path)
+                    log_file = "spack-build-%02d-%s-out.txt" % (
+                        i + 1, phase_name.lower()
+                    )
+                    log_file = os.path.join(log_dir, log_file)
+
+                    try:
+                        # DEBUGGING TIP - to debug this section, insert an IPython
+                        # embed here, and run the sections below without log capture
+                        log_contextmanager = log_output(
+                            log_file,
+                            self.echo,
+                            True,
+                            env=self.unmodified_env,
+                            filter_fn=self.filter_fn
+                        )
+
+                        with log_contextmanager as logger:
+                            with logger.force_echo():
+                                inner_debug_level = tty.debug_level()
+                                tty.set_debug(debug_level)
+                                tty.msg(
+                                    "{0} Executing phase: '{1}'" .format(
+                                        self.pre,
+                                        phase_name
+                                    )
+                                )
+                                tty.set_debug(inner_debug_level)
 
-                # Keep a log file for each phase
-                log_dir = os.path.dirname(pkg.log_path)
-                log_file = "spack-build-%02d-%s-out.txt" % (
-                    i + 1, phase_name.lower()
-                )
-                log_file = os.path.join(log_dir, log_file)
+                            # Redirect stdout and stderr to daemon pipe
+                            phase = getattr(pkg, phase_attr)
+                            self.timer.phase(phase_name)
 
-                try:
-                    # DEBUGGING TIP - to debug this section, insert an IPython
-                    # embed here, and run the sections below without log capture
-                    log_contextmanager = log_output(
-                        log_file,
-                        self.echo,
-                        True,
-                        env=self.unmodified_env,
-                        filter_fn=self.filter_fn
-                    )
+                            # Catch any errors to report to logging
+                            phase(pkg.spec, pkg.prefix)
+                            spack.hooks.on_phase_success(pkg, phase_name, log_file)
```

Comment:
Right now the post-install hooks are skipped on Windows. Can we invert this loop:

* Create a function to return windows logger or `log_output` objects
* The Windows logger can create a noop `force_echo` function

Reply:
I think this would make the code easier to maintain, but if the post install hooks should in fact not run, then I have overlooked that.

Reply:
This change was from #22177 (somewhere along the way the PR number was lost). I know this file in particular has had multiple merge conflicts so something may have been re-based incorrectly.